### PR TITLE
AGENT-264: validate agent-config macs

### PR DESCRIPTION
**Pull request by andfasano**
This patch adds two new validations for the agent-config file to ensure that the configured mac addresses are not duplicated.
Additionaly, tests have been refactored to improve (hopefully) the readability and to make it easier adding new cases.

### CODE REVIEW

File: pkg/asset/agent/agentconfig/agent_config.go [hunk 1]
```diff
@@ -150,81 +150,75 @@ func (a *AgentConfig) finish() error {
...
+			continue
 		}
 
-		for j := range node.Interfaces {
-			if node.Interfaces[j].MacAddress == "" {
-				macAddressPath := interfacePath.Index(j).Child("macAddress")
-				allErrs = append(allErrs, field.Required(macAddressPath, "each interface must have a MAC address defined"))
-			}
+		if _, ok := macs[mac]; ok {
+			allErrs = append(allErrs, field.Invalid(macAddressPath, "duplicated MAC address found", mac))
```
Comment (pawanpinjarkar) [inline pkg/asset/agent/agentconfig/agent_config.go:194]: `duplicate MAC address found`

**Review by pawanpinjarkar — COMMENTED** [review]
just a minor thing, otherwise /lgtm

**Comment by lranjbar** [comment]
/approve

I like how this is making the unit tests more descriptive. 👍🏼

**Pull request merged**